src/main/java/org/radarcns/biovotion/BiovotionDeviceManager.java

```diff
@@ -384,21 +673,67 @@ public void onStreamValueReceived(@NonNull final StreamValue unit) {
 
             case RawBoard:
                 RawBoard rawboard = (RawBoard) unit.unit;
-                deviceStatus.setTemperature(rawboard.localTemp);
-                deviceStatus.setTemperatureObject(rawboard.objectTemp);
-                deviceStatus.setTemperatureBaro(rawboard.barometerTemp);
-                deviceStatus.setGsrAmplitude(rawboard.gsrAmplitude);
-                deviceStatus.setGsrPhase(rawboard.gsrPhase);
+                deviceStatus.setTemperature(rawboard.objectTemp, rawboard.localTemp, rawboard.barometerTemp);
+                deviceStatus.setGalvanicSkinResponse(rawboard.gsrAmplitude, rawboard.gsrPhase);
+                float[] latestTemp = deviceStatus.getTemperatureAll();
+                float[] latestGSR = deviceStatus.getGalvanicSkinResponse();
 
                 BiovotionVSMTemperature tempValue = new BiovotionVSMTemperature((double) unit.timestamp, System.currentTimeMillis() / 1000d,
-                        deviceStatus.getTemperature(), deviceStatus.getTemperatureObject(), deviceStatus.getTemperatureBaro());
+                        latestTemp[0], latestTemp[1], latestTemp[2]);
                 BiovotionVSMGalvanicSkinResponse gsrValue = new BiovotionVSMGalvanicSkinResponse((double) unit.timestamp, System.currentTimeMillis() / 1000d,
-                        deviceStatus.getGsrAmplitude(), deviceStatus.getGsrPhase());
+                        latestGSR[0], latestGSR[1]);
 
                 dataHandler.addMeasurement(temperatureTable, deviceStatus.getId(), tempValue);
                 dataHandler.addMeasurement(gsrTable, deviceStatus.getId(), gsrValue);
                 break;
-        }
 
+            case RawAlgo:
+                RawAlgo rawalgo = (RawAlgo) unit.unit;
+                for (RawAlgo.RawAlgoUnit i : rawalgo.units) {
+                    //logger.info("Biovotion VSM RawAlgo: red:{} | green:{} | ir:{} | dark:{}", i.red, i.green, i.ir, i.dark);
+                    //logger.info("Biovotion VSM RawAlgo: x:{} | y:{} | z:{} | @:{}", i.x, i.y, i.z, (double) unit.timestamp);
+                    deviceStatus.setAcceleration(i.x, i.y, i.z);
+                    float[] latestAcc = deviceStatus.getAcceleration();
+
+                    BiovotionVSMAcceleration accValue = new BiovotionVSMAcceleration((double) unit.timestamp, System.currentTimeMillis() / 1000d,
+                            latestAcc[0], latestAcc[1], latestAcc[2]);
+
+                    // add measurements to a stack as long as new measurements are older. if a newer measurement is added, empty the stack into accelerationTable and start over
```

Comment:
Sounds fine. How do you ensure that the last piece of data gets sent, if there is no more request after that? Could that be part of your close method, or would you end up with a race condition?